ph5/utilities/dumpsegd.py

```diff
@@ -0,0 +1,188 @@
+    i = 1
+    for c in sd.reel_headers.external_header_shot:
+        print i
+        i += 1
+        print_container(c)
+
+
+def trace_headers(sd):
+    n = 1
+    print "*** Trace Header ***", n
+
+    sd.process_trace_headers()
+    print_container(sd.trace_headers.trace_header)
+    i = 1
+    for c in sd.trace_headers.trace_header_N:
+        print i
+        i += 1
+        print_container(c)
+
+    trace = sd.read_trace(sd.samples)
+    if 'fairprint' in os.environ:
+        for s in trace:
+            print s
+
+    n += 1
+    while True:
+        if sd.isEOF():
+            break
+        print "*** Trace Header ***", n
+        sd.process_trace_headers()
+        print_container(sd.trace_headers.trace_header)
+        i = 1
+        for c in sd.trace_headers.trace_header_N:
+            print i
+            i += 1
+            print_container(c)
+
+        n += 1
+        trace = sd.read_trace(sd.samples)
+        if 'fairprint' in os.environ:
+            for s in trace:
+                print s
+
+    print "There were {0} traces.".format(n)
+
+
+def read_manufacture_code(filename):
+    """ read byte 17 for manufacture code"""
+    f = open(filename, 'rb')
+    f.seek(16)
+    byte = f.read(1)
+    swap = True
+    if sys.byteorder == 'big':
+        swap = False
+    bin = construct.BitStruct("BIN",
+                              construct.BitField(
+                                  "field", 8, swapped=swap))
+    bcd = bin.parse(byte)['field']
+    if sys.byteorder == 'little':
+        bcd = construct.ULInt64("xxx").build(bcd)
+    else:
+        bcd = construct.UBInt64("xxx").build(bcd)
+    code = bcd_py.bcd2int(bcd, 0, 2)
+    f.close()
+    return code
+
+
+def get_segdreader():
+    """
+        get the segdreader from manufacture code infile
+        or from second argument
+    """
+    KNOWN_CODE = {20: (segdreader, 'FairField'),
+                  61: (segdreader_smartsolo, 'SmartSolo')}
+    req_code_list = ["%s for %s format" % (k, KNOWN_CODE[k][1])
+                     for k in KNOWN_CODE.keys()]
+    req_code_str = ("Please give the second argument either "
+                    ' or '.join(req_code_list))
+
+    manu = read_manufacture_code(sys.argv[1])
+    if manu in KNOWN_CODE.keys():
+        reader = KNOWN_CODE[manu][0]
+    else:
+        try:
+            manu = sys.argv[2]
+            if manu in KNOWN_CODE.keys():
+                reader = KNOWN_CODE[manu][0]
+            else:
+                print("The second argument {0} is not one of the known codes:"
+                      " {1}.\n{2}".format(manu, KNOWN_CODE.keys(),
+                                          req_code_str))
+                raise Exception
+        except IndexError:
+            print("The manufacture code {0} is not one of the known codes:"
+                  " {1}.\n{2}".format(manu, KNOWN_CODE.keys(), req_code_str))
+            raise Exception
+    return reader
+
+
+def main():
+    global RH, TH
```

Comment:
The global variables from the original code:
https://github.com/PIC-IRIS/PH5/blob/58093e2ff651fba14cdb765aef6b271eae4934d7/ph5/utilities/dumpfair.py#L120
We decided not trying to change the global variables in the original codes. You may think that it's a new line because of the '+' sign. But it is because I changed the file name from dumpfair to dumpsegd.

Reply:
Sounds good, thanks for the confirmation on this.